chalice/deploy/deployer.py

```diff
@@ -392,24 +392,42 @@ def build(self, config, stage_name):
                 name=function.name, handler_name=function.handler_string,
                 stage_name=stage_name)
             resources.append(resource)
-        for event_source in config.chalice_app.event_sources:
-            scheduled_event = self._create_event_model(
-                config, deployment, event_source, stage_name)
-            resources.append(scheduled_event)
+        event_resources = self._create_lambda_event_resources(
+            config, deployment, stage_name)
+        resources.extend(event_resources)
         if config.chalice_app.routes:
             rest_api = self._create_rest_api_model(
-                config, deployment, config.chalice_app, stage_name)
+                config, deployment, stage_name)
             resources.append(rest_api)
-        for s3_event in config.chalice_app.s3_events:
-            bucket_notification = self._create_bucket_notification(
-                config, deployment, s3_event, stage_name)
-            resources.append(bucket_notification)
         return models.Application(stage_name, resources)
 
+    def _create_lambda_event_resources(self, config, deployment, stage_name):
+        # type: (Config, models.DeploymentPackage, str) -> List[models.Model]
+        resources = []  # type: List[models.Model]
+        for event_source in config.chalice_app.event_sources:
+            if isinstance(event_source, app.S3EventConfig):
```

Comment:
I'm assuming this is one of the places you're looking to remove duplication?

Reply:
No, this wasn't one of the places I had in mind.  There's not much duplication other than sharing the appending to a list.  The only thing I could see changing in the future is replacing the if/else chain with dynamic dispatch, but 3 if's is ok with me for now.